### ISSUE TITLE
Tuple assignments on non-value storage variables

### ISSUE THREAD

**Issue by dddejan** [SRI-CSL/solidity#40]
```solidity 
pragma solidity >=0.5.0;

contract StructsSwap {

  struct A {
    int x;
  }

  A a1;
  A a2;

  function() external payable {
    a1.x = 1;
    a2.x = 2;
    assert(a1.x == 1);
    assert(a2.x == 2);

    (a1, a2) = (a2, a1);
    assert(a1.x == 1);
    assert(a2.x == 1);
  }
}
```

Passes truffle but running verifier results in:
```
solc-verify.py test/solc-verify/StructsSwap.sol 
StructsSwap::[fallback]: ERROR
 - test/solc-verify/StructsSwap.sol:19:5: Assertion might not hold.
StructsSwap::[implicit_constructor]: OK
Errors were found by the verifier.
Use --show-warnings to see the warnings for potential false alarms.
```

**Comment by dddejan** [comment]
@hajduakos what is the status of this? I was under the impression that storage references are already implemented. Is it just that semantics have not been updated to use references in tuple assignments?

**Comment by hajduakos** [comment]
Done in a1619c4a3bb3f067227141210449b8e3784156f1, will do some cleanup and then merge

**Comment by hajduakos** [comment]
Merged in b3d28ca, @dddejan please check and close
